pennylane/ops/qubit/state_preparation.py

```diff
@@ -156,6 +158,32 @@ def compute_decomposition(state, wires):
         """
         return [MottonenStatePreparation(state, wires)]
 
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return True
+
+    def matrix(self, wire_order=None):
+        """Returns a ket vector representing the state being created.
+
+        QSV([1/2, 1/2, 1/2, 1/2], wires=[0,2]) = 1/4 * (|0 0> + |0 1> + |1 0> + |1 1>)
+                                               = 1/4 * (|000> + |001> + |100> + |101>)
+        """
+        if wire_order is None:
+            return np.array(self.parameters[0], dtype="complex128")
+
+        wires = Wires(wire_order)
+        if not wires.contains_wires(self.wires):
+            raise WireError("Custom wire_order must contain all QubitStateVector wires")
+        num_wires = len(wires)
+        indices = [0] * num_wires
+        for base_wire_idx in [wires.index(w) for w in self.wires]:
+            indices[base_wire_idx] = slice(None)  # same as ":"
+
+        ket = np.zeros((2,) * num_wires)
+        ket[tuple(indices)] = np.reshape(self.parameters[0], (2,) * len(self.wires))
+        return ket.reshape(2**num_wires)
```

Comment:
@timmysilv I wonder if you can re-use either https://github.com/PennyLaneAI/pennylane/blob/2c595a67d541dab4da8461ce8b6336fe25ad298e/pennylane/math/matrix_manipulation.py#L27 or https://github.com/PennyLaneAI/pennylane/blob/2c595a67d541dab4da8461ce8b6336fe25ad298e/pennylane/utils.py#L252 here?

Reply:
(`expand_matrix` ideally since it is newer, and `expand_vector` I think is a bit forgotten)

Reply:
nice!

Reply:
A good time to unify `expand_matrix`, `expand_vector`, and the new one where possible.

Reply:
I'll take note of it, see how we can bring it all together outside of this PR